### PR TITLE
reference-manual: preloaded-image: update

### PR DESCRIPTION
**Pull request by munoz0raul**
Signed-off-by: Raul Muñoz <raul@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for bb698ecc55b4333732895f6d22580c39804df93c are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1124/docs/artifacts/html/index.html

File: source/reference-manual/linux/preloaded-images.rst [hunk 1]
```diff
@@ -138,6 +136,29 @@ container's master branch and the "devel" Target will have both
...
+
+    lmp:
+      tagging:
+        refs/heads/sec-fix
+        # produce a target with containers from master
+          - tag: sec-fix
+          inherit: master
+        # produce a target with containers from devel
+          - tag: sec-fix
+          inherit: devel
```
Comment (doanac) [inline source/reference-manual/linux/preloaded-images.rst:149]:
```suggestion
          # produce a target with containers from master
          - tag: sec-fix
            inherit: master
          # produce a target with containers from devel
          - tag: sec-fix
            inherit: devel
```

File: source/reference-manual/linux/preloaded-images.rst [hunk 2]
```diff
@@ -138,6 +136,29 @@ container's master branch and the "devel" Target will have both
...
+    ...
+   
+    containers:
+      tagging:
+    ...
+    ref_options:
+      refs/heads/sec-fix:
+        preloaded_images:
+          enabled: true
+          shortlist: "money-making-app,debug-tools"
```
Comment (doanac) [inline source/reference-manual/linux/preloaded-images.rst:159]: I don't think this is valid factory-config content. we'd need to update: https://git.foundries.io/development/cloudplatforms/infrastructure/repoz/-/blob/master/factory_definition.py#L52-55 with something like:
~~~
preloaded_images: Optional[PreloadedImages]
~~~

Comment (munoz0raul) [reply source/reference-manual/linux/preloaded-images.rst:159]: I didn't know the right way to do it as well. I read the documentation and tried to update it to what I thought was right.
If you could read the previous documentation and suggest the right way, I appreciate it.

File: source/reference-manual/linux/preloaded-images.rst [hunk 3]
```diff
@@ -122,7 +120,7 @@ User occasionally combine preloaded images with a certain kind of
 In this scenario the devel and master container branches may not even
 have the same set of apps/containers. It's generally recommended
 to not produce a preloaded image. However, a ``ref_option`` could
-be added to set ``ASSEMBLE_SYSTEM_IMAGE: "0"`` for that branch.
```
Comment (mike-sul) [inline source/reference-manual/linux/preloaded-images.rst:125]: The previous version is actually correct. In `ref_option` we use `ASSEMBLE_SYSTEM_IMAGE` and `APP_SHORTLIST` (as well as any other configuration param) to override default values per ref/branch. `ref_option.<ref>.params.` is kind of environment variables for a specific ref/branch.

File: source/reference-manual/linux/preloaded-images.rst [hunk 4]
```diff
@@ -24,12 +24,13 @@ The easiest way to configure this is updating a Factory's
...
+   preloaded_images:
+     enabled: true
 
-  # Optional: The list of apps to preload can be set globally with:
-  # params:
-  #   APP_SHORTLIST: "money-making-app,debug-tools"
+  # Optional: The list of apps to preload can be with:
+  # containers:
+  #   preloaded_images:
+  #     shortlist: "money-making-app,debug-tools" "money-making-app,debug-tools"
```
Comment (mike-sul) [inline source/reference-manual/linux/preloaded-images.rst:33]: `enabled: true` must be between `containers:` and `preloaded_images`.

File: source/reference-manual/linux/preloaded-images.rst (hunk 2 above)
Comment (mike-sul) [reply source/reference-manual/linux/preloaded-images.rst:159]:
```
ref_options:
      refs/heads/sec-fix:
        params:
          APP_SHORTLIST: "money-making-app,debug-tools"
```

**Comment by mike-sul** [comment]
I think there is incorrectness in the current version.

```
APP_SHORTLIST will pick up its override value from the “refs/heads/sec-fix” ref_option. If devel and master had different apps such as:

devel: fiotest,moneymaking-app,debug-tools
master: moneymaking-app
```

The previous example sets just `moneymaking-app,debug-tools` for `devel` branch. And the following sentence states that the union of apps is `"money-making-app,debug-tools"`.

**Comment by munoz0raul** [comment]
Add a complete new PR with more changes:
https://github.com/foundriesio/docs/pull/275

**Pull request closed without merging**